File: bank_frontend/src/app/modules/bookstore/model/credit-request.ts
import { EmploymentInfo } from "./employmentInfo";

export class CreditRequest {
    clientUsername: string = '';
    totalAmount: number = 0;
    minPaymentDeadline: string = '';
    maxPaymentDeadline: string = '';
    numOfInstallments: number = 0;
    employmentInfo: EmploymentInfo = new EmploymentInfo();
  
    public constructor(obj?: any) {
      if(obj) {
        this.clientUsername = obj.clientUsername;
        this.totalAmount = obj.totalAmount;
        this.minPaymentDeadline = obj.minPaymentDeadline;
        this.maxPaymentDeadline = obj.maxPaymentDeadline;
        this.numOfInstallments = obj.numOfInstallments;
        this.employmentInfo = obj.employmentInfo;
      }
    }
  }

File: bank_backend/src/main/java/com/bank/sbnz/model/CreditRequest.java
package com.bank.sbnz.model;

import com.bank.sbnz.converter.EmploymentInfoConverter;
import com.bank.sbnz.enums.CreditRequestStatus;
import jakarta.persistence.*;
import lombok.*;

import java.time.LocalDate;

@AllArgsConstructor
@NoArgsConstructor
@Entity
@Builder
@Getter
@Setter
@ToString
@Table(name = "credit_request")
public class CreditRequest {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    @Column(name = "id", nullable = false)
    private Long id;
    @ManyToOne(fetch = FetchType.EAGER, cascade = CascadeType.ALL)
    @JoinColumn(name = "client_id")
    private AppUser client;
    @Column
    private double totalAmount;
    @Column
    private LocalDate minPaymentDeadline;
    @Column
    private LocalDate maxPaymentDeadline;
    @Column
    private CreditRequestStatus status;
    @Convert(converter = EmploymentInfoConverter.class)
    private EmploymentInfo employmentInfo;
    @Column
    private int numOfInstallments;
    @Column
    private double oneInstallmentAmount;
    @Column
    private LocalDate nextPaymentDate; //if approved
}
